src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -301,26 +378,68 @@ public int getIdleMinutes() {
         return (idleMinutes != null) ? idleMinutes : 0;
     }
 
+    @DataBoundSetter
+    public void setIdleMinutes(Integer idleMinutes) {
+        this.idleMinutes = (idleMinutes != null) ? idleMinutes : 0;
+    }
+
     public int getMaxSize() {
         return maxSize;
     }
 
+    @DataBoundSetter
+    public void setMaxSize(int maxSize) {
+        this.maxSize = maxSize;
+    }
+
     public int getMinSize() {
         return minSize;
     }
 
-    public int getMinSpareSize() {
+    @DataBoundSetter
+    public void setMinSize(int minSize) {
+        if (minSize < 0) {
+            warning("Cloud parameter 'minSize' can't be less than 0, setting to 0");
+        }
+        this.minSize = Math.max(0, minSize);
+    }
+
+    public synchronized int getMinSpareSize() {
         return minSpareSize;
     }
 
+    @DataBoundSetter
+    public synchronized void setMinSpareSize(int minSpareSize) {
+        this.minSpareSize = Math.max(0, minSpareSize);
+    }
+
+    public int getMaxTotalUses() {
```

Comment:
This getter can simply return `maxTotalUses` since with your setter handling nulls. 

Same comment for `getNumExecutors`